README.md

```diff
@@ -100,9 +123,87 @@ act_when($isChuckNorrisMighty, new ChuckNorris($data));
 act_unless($isChuckNorrisPuny, new ChuckNorris($data));
 ```
 
+### Dependency Injection
+
+You can even injection actions as a dependency with your app!
+
+```php
+use Kirschbaum\Actions\Contracts\Actionable;
+
+public function index (Actionable $action)
+{
+    $action->act(new ChuckNorris($data));
+    $action->actWhen($isChuckNorrisMighty, new ChuckNorris($data));
+    $action->actUnless($isChuckNorrisPuny, new ChuckNorris($data));
+}
+```
+
+## Handling Failures
+
+We all know Chuck Norris isn't going to fail us, but he isn't the only one using this... Handling failures is pretty easy with actions. Out of the box, any exceptions get handled by Laravel's exception handler. If you'd rather implement your own logic during a failure, add a `failed()` method to your action. It's that easy!
+
+```php
+/**
+ * Handle failure of the action.
+ *
+ * @throws Throwable
+ *
+ * @return mixed
+ */
+public function failed(Throwable $exception)
+{
+    event(new VanDammeFailedEvent);
+}
+```
+
+Hashtag #BAM!
+
+### Custom Exceptions
+
+Another option for handling failures is to tell the action to fire its own exception. If you don't need the extra overhead of writing your own `failed()` method, you can just tell your action to throw a custom exception. It's as simple as just defining the exception you want thrown from the action.
+
+```php
+/**
+ * Event to dispatch if action throws an exception.
+ *
+ * @var string
+ */
+public $exception = SeagalFailedException::class;
+```
+
+## Testing
+
+Have no fear. Testing all of this is very straight forward. There are two approaches to testing built in.
+
+### Testing Facades
+
+If you are using Facades to implement your actions, you can use the standard 'shouldReceive()' method directly from the Facade.
+
+```php
+use Kirschbaum\Actions\Facades\Action;
+
+Action::shouldReceive('act')
+    ->once()
+    ->andReturnTrue();
+```
+
+### Mocking
+
+If you are using helpers, the `CanAct` trait, or dependency injection, you can easily mock the `Actionable` interface with Laravel's mocking tools.
+
+```php
+use Kirschbaum\Actions\Contracts\Actionable;
+
+$this->mock(Actionable::class, function ($mock) {
+    $mock->shouldReceive('act')
+        ->once()
+        ->andReturnTrue();
+});
+```
+
 ## Last thoughts
 
-If for some reason you'd prefer not to use the cool eventing system, that's fine. Just call your action like this:
+If for some reason you'd prefer not to use the cool eventing system, facades, mocking, etc., that's fine. Just call your action like this:
```

Comment:
You used name nomenclature on Facades earlier, so I would try to maintain that throughout the document
```suggestion
If for some reason you'd prefer not to use the cool Eventing System, Facades, Mocking, etc., that's fine. Just call your action like this:
```